src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowRecordConverter.java

```diff
@@ -0,0 +1,138 @@
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Map;
+
+/**
+ * Utility class for converting the record from ServiceNow data type to CDAP schema data types
+ */
+public class ServiceNowRecordConverter {
+  private static final String DATE_PATTERN = "yyyy-MM-dd";
+  private static final String DATE_TIME_PATTERN = "yyyy-MM-dd HH:mm:ss";
+  private static final String TIME_PATTERN = "HH:mm:ss";
+
+  public static void convertToValue(String fieldName, Schema fieldSchema, Map<String, String> record,
+                                    StructuredRecord.Builder recordBuilder) {
+    String fieldValue = record.get(fieldName);
+    if (fieldValue == null || fieldValue.isEmpty()) {
+      // Set 'null' value as it is
+      recordBuilder.set(fieldName, null);
+      return;
+    }
+
+    fieldSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable() : fieldSchema;
+    Schema.LogicalType fieldLogicalType = fieldSchema.getLogicalType();
+    // Get values of logical types properly
+    if (fieldLogicalType != null) {
+      switch (fieldLogicalType) {
+        case DATETIME:
+          DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(DATE_TIME_PATTERN);
+          try {
+            recordBuilder.setDateTime(fieldName, LocalDateTime.parse(fieldValue, dateTimeFormatter));
+          } catch (DateTimeParseException exception) {
+            throw new UnexpectedFormatException(
+              String.format("Field '%s' of type '%s' with value '%s' is not in ISO-8601 format.",
+                            fieldName, fieldSchema.getDisplayName(), fieldValue), exception);
+          }
+          return;
+        case DATE:
+          DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern(DATE_PATTERN);
+          try {
+            recordBuilder.setDate(fieldName, LocalDate.parse(fieldValue, dateFormatter));
+          } catch (DateTimeParseException exception) {
+            throw new UnexpectedFormatException(
+              String.format("Field '%s' of type '%s' with value '%s' is not in ISO-8601 format.",
+                            fieldName, fieldSchema.getDisplayName(), fieldValue), exception);
+          }
+          return;
+        case TIME_MILLIS:
+          DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern(TIME_PATTERN);
+          try {
+            recordBuilder.setTime(fieldName, LocalTime.parse(fieldValue, timeFormatter));
+          } catch (DateTimeParseException exception) {
+            throw new UnexpectedFormatException(
+              String.format("Field '%s' of type '%s' with value '%s' is not in ISO-8601 format.",
+                            fieldName, fieldSchema.getDisplayName(), fieldValue), exception);
+          }
+          return;
+        default:
+          throw new UnexpectedFormatException(String.format("Field '%s' is of unsupported type '%s'", fieldName,
```

Comment:
change to IllegalStateException to make it consistent with the exception that is thrown when the physical type switch statement

Reply:
Changed to `IllegalStateException` cbfb212c77242fe61b2ef14c690a9079b6a9e22f